atuin-client/src/history.rs

```diff
@@ -5,38 +5,57 @@ use serde::{Deserialize, Serialize};
 
 use atuin_common::utils::uuid_v7;
 
-// Any new fields MUST be Optional<>!
+mod builder;
+
+/// A marker type used to seal the `History` struct, preventing it from being constructed directly.
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub(self) struct HistorySeal;
```

Comment:
Created `HistorySeal` ZST, which can be constructed from this module, but is unavailable for any external ones. This assures that now only builders can be used to created `History` struct, and these are type-safe, so incorrect `History` object cannot be produced.